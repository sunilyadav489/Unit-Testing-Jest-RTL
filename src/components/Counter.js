import React, {useState} from 'react';

const Counter = () => {
    let [counter, setCounter] = useState(1);

    const increamentCounter = () => {
        setCounter(counter+1);
    }

    const decreamentCounter = () => {
        setCounter(counter-1);
    }

  return (
    <>
        <h4>Counter Component (Testing Events and Snapshot)</h4>
        <div className='counterWrapper'>
            <button data-testid="increment" onClick={increamentCounter}>+</button>
            <p data-testid="counter">{counter}</p>
            <button data-testid="decrement" onClick={decreamentCounter}>-</button>
        </div>
    </>
    
  )
}

export default Counter