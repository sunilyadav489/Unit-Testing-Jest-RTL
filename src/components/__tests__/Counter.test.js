import {render, screen, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

import Counter from '../Counter';

describe('Counter', () => {
    it('Should increment the counter' , () => {
        render(<Counter/>);
        const counter = screen.getByTestId("counter");
        const increment = screen.getByTestId('increment');
        fireEvent.click(increment);
        expect(counter).toHaveTextContent(2);
    })

    it('Should decrement the counter', () => {
        render(<Counter/>);
        const counter = screen.getByTestId("counter");
        const decrement = screen.getByTestId('decrement');
        fireEvent.click(decrement);
        expect(counter).toHaveTextContent(0);
    })
});

//Use case : To validate your UI does not change unexpectedly
describe('Counter snapshot', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
})