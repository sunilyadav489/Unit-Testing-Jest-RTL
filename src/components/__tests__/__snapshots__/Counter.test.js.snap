// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <h4>
    Counter Component (Testing Events and Snapshot)
  </h4>,
  <div
    className="counterWrapper"
  >
    <button
      data-testid="increment"
      onClick={[Function]}
    >
      +
    </button>
    <p
      data-testid="counter"
    >
      1
    </p>
    <button
      data-testid="decrement"
      onClick={[Function]}
    >
      -
    </button>
  </div>,
]
`;
